### PR TITLE
Admin Users API

### PR DESCRIPTION
**Pull request by omarismail**
## Description

This PR adds all the [Admin Users API](https://www.terraform.io/docs/cloud/api/admin/users.html).

It implements: 
```
type AdminUsers interface {
	// List all the users of the given installation.
	List(ctx context.Context, options AdminUserListOptions) (*AdminUserList, error)

	// Delete a user by its ID.
	Delete(ctx context.Context, userID string) error

	// Suspend a user account by its ID.
	Suspend(ctx context.Context, userID string) (*AdminUser, error)

	// Unsuspend a user account by its ID.
	Unsuspend(ctx context.Context, userID string) (*AdminUser, error)

	// GrantAdminPrivlages to a user account by its ID.
	GrantAdminPrivlages(ctx context.Context, userID string) (*AdminUser, error)

	// RevokeAdminPrivlages to a user account by its ID.
	RevokeAdminPrivlages(ctx context.Context, userID string) (*AdminUser, error)

	// Disable2FA disables a user's two-factor authentication in the situation
	// where they have lost access to their device and recovery codes.
	Disable2FA(ctx context.Context, userID string) (*AdminUser, error)

	// ImpersonateUser allows an admin to begin a new session as another user in
	// the system
	ImpersonateUser(ctx context.Context, userID string, options ImpersonateReasonOption) error

	// UnimpersonateUser allows an admin to end an impersonationn session of
	// another user in the system
	UnimpersonateUser(ctx context.Context, userID string) error
}
```

## Testing plan

1. Pull down and run the tests 

## Output from tests

```
$ TFE_TOKEN=<token> TFE_ADDRESS=<address>  ENABLE_TFE=1 go test -v ./...  -timeout=30m
...
PASS
ok  	github.com/hashicorp/go-tfe	514.436s
?   	github.com/hashicorp/go-tfe/examples/organizations	[no test files]
?   	github.com/hashicorp/go-tfe/examples/workspaces	[no test files]
```

## External links

- [API documentation](https://www.terraform.io/docs/cloud/api/admin/users.html)

### CODE REVIEW

**Review by chrisarcand — CHANGES_REQUESTED** [review]
Nice!

File: admin_user.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+	Delete(ctx context.Context, userID string) error
+
+	// Suspend a user by its ID.
+	Suspend(ctx context.Context, userID string) (*AdminUser, error)
+
+	// Unsuspend a user by its ID.
+	Unsuspend(ctx context.Context, userID string) (*AdminUser, error)
+
+	// GrantAdminPrivlages grants admin pirvlages to a user by its ID.
+	GrantAdminPrivlages(ctx context.Context, userID string) (*AdminUser, error)
```
[the remark this answers is not in the document]
Comment (omarismail) [reply admin_user.go:31]: Sounds good

File: admin_user.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+	// where they have lost access to their device and recovery codes.
+	Disable2FA(ctx context.Context, userID string) (*AdminUser, error)
+
+	// ImpersonateUser allows an admin to begin a new session as another user in
+	// the system.
+	ImpersonateUser(ctx context.Context, userID string, options ImpersonateReasonOption) error
+
+	// UnimpersonateUser allows an admin to end an impersonationn session of
+	// another user in the system.
+	UnimpersonateUser(ctx context.Context, userID string) error
```
[the remark this answers is not in the document]
Comment (omarismail) [reply admin_user.go:46]: Yeeaahh, I was iffy on this one. It didn't make sense in the API but it was there, so figure'd i'd add it. 

I actually failed the read the entire doc area 🤦‍♂️ . 

I read the blue highlighted area and thought it was workable via the API, even though when I tested it it seemed to work under certain user conditions. But I didn't read the "Important" yellow area.

Will remove

image

[the remark this answers is not in the document]
Comment (omarismail) [reply admin_user.go]: Updated

File: admin_user.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+	// Suspend a user by its ID.
+	Suspend(ctx context.Context, userID string) (*AdminUser, error)
+
+	// Unsuspend a user by its ID.
+	Unsuspend(ctx context.Context, userID string) (*AdminUser, error)
+
+	// GrantAdmin grants admin privilages to a user by its ID.
+	GrantAdmin(ctx context.Context, userID string) (*AdminUser, error)
+
+	// RevokeAdmin revokees admin privlages to a user by its ID.
```
Comment (chrisarcand) [inline admin_user.go:33]: 😜 

```suggestion
	// RevokeAdmin revokees admin privilages to a user by its ID.
```

File: admin_user.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+	au := &AdminUser{}
+	err = a.client.do(ctx, req, au)
+	if err != nil {
+		return nil, err
+	}
+
+	return au, nil
+}
+
+// RevokeAdmin revokes admin privlages to a user by its ID.
```
Comment (chrisarcand) [inline admin_user.go:181]:
```suggestion
// RevokeAdmin revokes admin privilages to a user by its ID.
```

File: admin_user_test.go [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+	})
+
+	t.Run("an non-existing user", func(t *testing.T) {
+		err := client.Admin.Users.Delete(ctx, "non-existing-user-id")
+		require.Error(t, err)
+	})
+}
+
+func TestAdminUsers_AdminPrivlages_Suspensions(t *testing.T) {
```
Comment (chrisarcand) [inline admin_user_test.go:142]:
```suggestion
func TestAdminUsers_AdminPrivilages_Suspensions(t *testing.T) {
```

**Review by chrisarcand — CHANGES_REQUESTED** [review]

File: admin_user.go (hunk 4 above)
Comment (omarismail) [reply admin_user.go:181]: Clearly I can't spell

**Review by chrisarcand — APPROVED** [review]
Looks great! Nice work! 🚢

**Pull request merged**